streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -148,132 +158,99 @@ public boolean isValidTransition(final State newState) {
             this.time = time;
-            this.requestTimeOut = requestTimeOut;
         }
 
         @Override
         public void onPartitionsAssigned(final Collection<TopicPartition> assignment) {
             log.debug("{} at state {}: new partitions {} assigned at the end of consumer rebalance.\n" +
-                    "\tassigned active tasks: {}\n" +
-                    "\tassigned standby tasks: {}\n" +
-                    "\tcurrent suspended active tasks: {}\n" +
-                    "\tcurrent suspended standby tasks: {}\n" +
-                    "\tprevious active tasks: {}",
-                logPrefix,
-                state,
-                assignment,
-                partitionAssignor.activeTasks().keySet(),
-                partitionAssignor.standbyTasks().keySet(),
-                suspendedTasks.keySet(),
-                suspendedStandbyTasks.keySet(),
-                prevActiveTasks);
+                              "\tassigned active tasks: {}\n" +
+                              "\tassigned standby tasks: {}\n" +
+                              "\tcurrent suspended active tasks: {}\n" +
+                              "\tcurrent suspended standby tasks: {}\n",
+                      logPrefix,
+                      state,
+                      assignment,
+                      partitionAssignor.activeTasks().keySet(),
+                      partitionAssignor.standbyTasks().keySet(),
+                      active.previousTasks(),
+                      standby.previousTasks());
 
             final long start = time.milliseconds();
             try {
-                storeChangelogReader = new StoreChangelogReader(getName(), restoreConsumer, time, requestTimeOut);
-                setStateWhenNotInPendingShutdown(State.ASSIGNING_PARTITIONS);
+
+                if (!setStateWhenNotInPendingShutdown(State.ASSIGNING_PARTITIONS)) {
+                    return;
+                }
                 // do this first as we may have suspended standby tasks that
                 // will become active or vice versa
                 closeNonAssignedSuspendedStandbyTasks();
                 closeNonAssignedSuspendedTasks();
-                addStreamTasks(assignment, start);
-                storeChangelogReader.restore();
+                addStreamTasks(assignment);
                 addStandbyTasks(start);
                 streamsMetadataState.onChange(partitionAssignor.getPartitionsByHostState(), partitionAssignor.clusterMetadata());
-                lastCleanMs = time.milliseconds(); // start the cleaning cycle
-                setStateWhenNotInPendingShutdown(State.RUNNING);
+                storeChangelogReader.reset();
+                Set<TopicPartition> partitions = active.uninitializedPartitions();
+                log.trace("{} pausing partitions: {}", logPrefix, partitions);
+                consumer.pause(partitions);
+                setStateWhenNotInPendingShutdown(State.PARTITIONS_ASSIGNED);
```

Comment:
Ack, my bad. The question then is: could we just rename `ASSIGNING_PARTITIONS` to `PARTITIONS_ASSIGNED` since the former now becomes a very transient stage?